sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/StringBasedCosmosQuery.java

```diff
@@ -49,13 +52,29 @@ public Object execute(final Object[] parameters) {
         final CosmosParameterAccessor accessor = new CosmosParameterParameterAccessor(getQueryMethod(), parameters);
         final ResultProcessor processor = getQueryMethod().getResultProcessor().withDynamicProjection(accessor);
 
-        List<SqlParameter> sqlParameters = getQueryMethod().getParameters().stream()
-                            .filter(p -> !Pageable.class.isAssignableFrom(p.getType()) && !Sort.class.isAssignableFrom(p.getType()))
-                            .map(p -> new SqlParameter("@" + p.getName().orElse(""),
-                                                       toCosmosDbValue(parameters[p.getIndex()])))
-                            .collect(Collectors.toList());
+        String expandedQuery = query;
+        List<SqlParameter> sqlParameters = new ArrayList<>();
+        for (int paramIndex = 0; paramIndex < parameters.length; paramIndex++) {
+            Parameter queryParam = getQueryMethod().getParameters().getParameter(paramIndex);
+            if (parameters[paramIndex] instanceof Collection) {
+                ArrayList<String> expandParam = (ArrayList<String>) ((Collection<?>) parameters[paramIndex]).stream()
```

Comment:
nit: We should not strictly cast it to ArrayList<String> here, instead keep it open to generic `List<String>` here. 

```
List<String> expandParam = ((Collection<?>) parameters[paramIndex]).stream().map(Object::toString).collect(Collectors.toList());
```

Reply:
Done.